The CI failures are real:

```
$ /opt/hostedtoolcache/Python/3.12.13/x64/bin/python -P -c 'import numpy'
  Traceback (most recent call last):
    File "<string>", line 1, in <module>
    File "/home/runner/work/numpy/numpy/build-install/usr/lib/python3.12/site-packages/numpy/__init__.py", line 454, in <module>
      from . import lib, matrixlib as _mat
    File "/home/runner/work/numpy/numpy/build-install/usr/lib/python3.12/site-packages/numpy/lib/__init__.py", line 18, in <module>
      from . import (
    File "/home/runner/work/numpy/numpy/build-install/usr/lib/python3.12/site-packages/numpy/lib/_arraypad_impl.py", line 10, in <module>
      from numpy.lib._index_tricks_impl import ndindex
    File "/home/runner/work/numpy/numpy/build-install/usr/lib/python3.12/site-packages/numpy/lib/_index_tricks_impl.py", line 8, in <module>
      import numpy.matrixlib as matrixlib
    File "/home/runner/work/numpy/numpy/build-install/usr/lib/python3.12/site-packages/numpy/matrixlib/__init__.py", line 4, in <module>
      from . import defmatrix
    File "/home/runner/work/numpy/numpy/build-install/usr/lib/python3.12/site-packages/numpy/matrixlib/defmatrix.py", line 13, in <module>
      from numpy.linalg import matrix_power
    File "/home/runner/work/numpy/numpy/build-install/usr/lib/python3.12/site-packages/numpy/linalg/__init__.py", line 87, in <module>
      from . import _linalg
    File "/home/runner/work/numpy/numpy/build-install/usr/lib/python3.12/site-packages/numpy/linalg/_linalg.py", line 3407, in <module>
      @array_function_dispatch(_tensordot_dispatcher)
       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    File "/home/runner/work/numpy/numpy/build-install/usr/lib/python3.12/site-packages/numpy/_core/overrides.py", line 148, in decorator
      verify_matching_signatures(implementation, dispatcher)
    File "/home/runner/work/numpy/numpy/build-install/usr/lib/python3.12/site-packages/numpy/_core/overrides.py", line 99, in verify_matching_signatures
      raise RuntimeError(f'implementation and dispatcher for {implementation} have '
  RuntimeError: implementation and dispatcher for <function tensordot at 0x7f40c8fbb4c0> have different function signatures
```

--- a/numpy/_core/numeric.py
+++ b/numpy/_core/numeric.py
@@ -1019,6 +1019,7 @@ def tensordot(a, b, axes=2):
           second to `b`. Both elements array_like must be of the same length.
           Each axis may appear at most once; repeated axes are not allowed.
           For example, ``axes=([1, 1], [0, 0])`` is invalid.
+
     Returns
     -------
     output : ndarray

--- a/numpy/linalg/_linalg.py
+++ b/numpy/linalg/_linalg.py
@@ -3375,13 +3375,14 @@ def matmul(x1, x2, /):
 
 # tensordot
 
-def _tensordot_dispatcher(x1, x2, /, *, axes=None):
-    return (x1, x2)
+
+def _tensordot_dispatcher(a, b, /, *, axes=None):
+    return (a, b)
 
 
 @array_function_dispatch(_tensordot_dispatcher)
-def tensordot(x1, x2, /, *, axes=2):
-    return _core_tensordot(x1, x2, axes=axes)
+def tensordot(a, b, /, *, axes=2):
+    return _core_tensordot(a, b, axes=axes)
 
 
 tensordot.__doc__ = _core_tensordot.__doc__

